src/basic/stat-util.c

```diff
@@ -471,3 +471,154 @@ int statx_fallback(int dfd, const char *path, int flags, unsigned mask, struct s
 
         return 0;
 }
+
+int fs_type_from_string(const char *name, uint32_t *ret) {
+        assert(name);
+        assert(ret);
+
+        if (streq(name, "apparmorfs"))
```

Comment:
can't we generate this automatically from linux/magic.h (plus a few manual additions)? or at least have some tool that validates on build that everything in linux/magic.h is also listed here?

also, please use gperf for this, see errno-from-gperf.h and such

Reply:
I'll have a look.